.github/workflows/core-dependency-update.yml

```
@@ -12,7 +12,72 @@ concurrency:
+  core-tests:
+    runs-on: ubuntu-latest
+    name: Run Core Provider Tests
+    
+    steps:
+      # Checkout repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      # Set up Go environment
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)